linkerd/app/core/src/control.rs

```diff
@@ -54,16 +58,46 @@ impl Config {
         B::Error: Into<Error> + Send + Sync,
         L: Clone + Param<tls::client::Config> + Send + 'static,
     {
-        let backoff = {
+        let connect_backoff = {
             let backoff = self.connect.backoff;
             move |_| Ok(backoff.stream())
         };
+
+        // When a DNS resolution fails, log the error and use the TTL, if there
+        // is one, to drive re-resolution attempts.
+        let resolve_backoff = {
+            let backoff = self.connect.backoff;
+            move |error: Error| {
+                warn!(%error, "Failed to resolve control-plane component");
+                if let Some(e) = error.downcast_ref::<dns::ResolveError>() {
+                    if let dns::ResolveErrorKind::NoRecordsFound {
+                        negative_ttl: Some(ttl_secs),
+                        ..
+                    } = e.kind()
```

Comment:
should we also log the TTL in this case?

Reply:
i don't think it's particularly relevant for the warning scope; and it'll generally be pretty obvious based on the interval from the debug perspective